ampersand-state.js

```diff
@@ -762,7 +763,8 @@ function extend(protoProps) {
         var omitFromExtend = [
             'dataTypes', 'props', 'session', 'derived', 'collections', 'children'
         ];
-        args.forEach(function processArg(def) {
+        for(var i = 0; i < arguments.length; i++) {
```

Comment:
This change has the biggest performance gains - see https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#32-leaking-arguments